Datasets: Add prompt template, clean up language, improve best practices (#136)
rungalileo/docs-official: concepts/datasets.mdx

### Version Management and History

One of the benefits of Galileo's dataset management is automatic versioning. Automatic versioning allows you to track how your test suite evolves over time as well as ensures reproducibility of your experiments. You can always reference specific versions of a dataset or work with the latest version:

<CodeGroup>
  <SnippetPythonGetDataset />
  <SnippetTSGetDataset />
</CodeGroup>

## Accessing Dataset Variables in Prompt Templates

> When you use datasets in Galileo, each sample in your dataset is made available to your prompt template as the `input` object.  This allows you to create dynamic prompts that adapt to the adata in each sample.

Reviewer: ```suggestion
When you use datasets in Galileo, each sample in your dataset is made available to your prompt template as the `input` object.  This allows you to create dynamic prompts that adapt to the data in each sample.
```

### Example Dataset
Suppose you have the following dataset: 

```
test_data = [
    {"city": "Rome, Italy", "metadata": {"days": 5}},
    {"city": "Tokyo, Japan", "metadata": {"days": 3}},
```

### Example Prompt Template
To reference fields from your dataset in your prompt, use double curly braces and the `input` object:

```
Plan a {{ input.metadata.days }}-day travel itinerary for a trip to {{ input.city }}.
Include daily sightseeing activities, dining suggestions, and local experiences.
```

- `{{ input.city }}` will be replaced with the value of the `city` field from each sample. 

> - `{{ input.metadata.days }}` will be replaced with teh value of the `days` field inside the `metadata` dictionary. 

Reviewer: ```suggestion
- `{{ input.metadata.days }}` will be replaced with the value of the `days` field inside the `metadata` dictionary. 
```

### How It works 

> For each sample in your dataset, Galileo will render the prompt template, replacing the variables iwth teh corresponding values from the sample. 

Reviewer: ```suggestion
For each sample in your dataset, Galileo will render the prompt template, replacing the variables with the corresponding values from the sample. 
```

> If a field is issing in a sample, the variable will be empty or may cause an error, so ensure your dataset is consistent. 

Reviewer: ```suggestion
If a field is missing in a sample, the variable will be empty or may cause an error, so ensure your dataset is consistent. 
```

3. Make sure fixes don't cause new problems
4. Build a library of test cases for future testing

This can be done either through the UI or programmatically, depending on your workflow.

## Best Practices for Dataset Management
When working with datasets consider these tips:

1. **Start Small and Representative**
  - **Why**: Beginning with a core set of representative test cases helps you quickly validate your workflow and catch obvious issues before scaling up.

>   - **How**: Select a handful of diverse, meaningful exampls that reflect the range of inputs your model will see.

Reviewer: ```suggestion
  - **How**: Select a handful of diverse, meaningful examples that reflect the range of inputs your model will see.
```
